src/stdlib_linalg.fypp

```diff
@@ -108,4 +201,200 @@ contains
         end do
       end function trace_${t1[0]}$${k1}$
     #:endfor
+
+
+    #:for k1, t1 in RCI_KINDS_TYPES
+      pure function is_square_${t1[0]}$${k1}$(A) result(res)
+        ${t1}$, intent(in) :: A(:,:)
+        logical :: res
+        res = (size(A,1) == size(A,2))
+      end function is_square_${t1[0]}$${k1}$
+    #:endfor
+
+
+    #:for k1, t1 in RCI_KINDS_TYPES
+      pure function is_diagonal_${t1[0]}$${k1}$(A) result(res)
+        ${t1}$, intent(in) :: A(:,:)
+        logical :: res
+        ${t1}$ :: zero
+        integer :: m, n, o, i, j
+        zero = 0 !zero of relevant type
```

Comment:
Why is `zero` not a parameter? Using parameter may allow some optimizations.

```suggestion
        ${t1}$, parameter :: zero = 0 !zero of relevant type
        integer :: m, n, o, i, j
```

Reply:
Thanks for the suggestion, it just didn't cross my mind. I've changed all three instances.